sharepoint-server-farm-ha/azuredeploy.json

```diff
@@ -0,0 +1,1985 @@
+			},
+			"defaultValue":"10.0.0.6"
+		},
+		"adVMPrefix": {
+			"type": "string",
+			"metadata": {
+				"Description": "The Prefix of the AD VM name"
+			},
+			"defaultValue":"dc"
+		},
+		"sqlVMPrefix": {
+			"type": "string",
+			"metadata": {
+				"Description": "The Prefix of the SQL VM Name"
+			},
+			"defaultValue":"sql"
+		},
+		"spVMPrefix": {
+			"type": "string",
+			"metadata": {
+				"Description": "The Prefix of the SP VM Name"
+			},
+			"defaultValue":"sps"
+		},
+		"adminUsername": {
+			"type": "string",
+			"metadata": {
+				"Description": "The name of the Administrator of the new VMs and Domain"
+			},
+			"defaultValue":"spAdministrator"
+		},
+		"adminPassword": {
+			"type": "securestring",
+			"metadata": {
+				"Description": "The password for the Administrator account of the new VMs and Domain"
+			}
+		},
+		"adVMSize": {
+			"type": "string",
+			"allowedValues": [
+				"Standard_D1",
+				"Standard_DS1",
+				"Standard_D2",
+				"Standard_DS2",
+				"Standard_D3",
+				"Standard_DS3",
+				"Standard_D4",
+				"Standard_DS4",
+				"Standard_D11",
+				"Standard_DS11",
+				"Standard_D12",
+				"Standard_DS12",
+				"Standard_D13",
+				"Standard_DS13",
+				"Standard_D14",
+				"Standard_DS14"
+			],
+			"metadata": {
+				"Description": "The size of the AD VMs Created"
+			},
+			"defaultValue": "Standard_DS1"
+		},
+		"sqlVMSize": {
+			"type": "string",
+			"allowedValues": [
+				"Standard_D1",
+				"Standard_DS1",
+				"Standard_D2",
+				"Standard_DS2",
+				"Standard_D3",
+				"Standard_DS3",
+				"Standard_D4",
+				"Standard_DS4",
+				"Standard_D11",
+				"Standard_DS11",
+				"Standard_D12",
+				"Standard_DS12",
+				"Standard_D13",
+				"Standard_DS13",
+				"Standard_D14",
+				"Standard_DS14"
+			],
+			"metadata": {
+				"Description": "The size of the SQL VMs Created"
+			},
+			"defaultValue": "Standard_DS3"
+		},
+		"witnessVMSize": {
+			"type": "string",
+			"allowedValues": [
+				"Standard_D1",
+				"Standard_DS1",
+				"Standard_D2",
+				"Standard_DS2",
+				"Standard_D3",
+				"Standard_DS3",
+				"Standard_D4",
+				"Standard_DS4",
+				"Standard_D11",
+				"Standard_DS11",
+				"Standard_D12",
+				"Standard_DS12",
+				"Standard_D13",
+				"Standard_DS13",
+				"Standard_D14",
+				"Standard_DS14"
+			],
+			"metadata": {
+				"Description": "The size of the Witness VM Created"
+			},
+			"defaultValue": "Standard_DS1"
+		},
+		"spVMSize": {
+			"type": "string",
+			"allowedValues": [
+				"Standard_D1",
+				"Standard_DS1",
+				"Standard_D2",
+				"Standard_DS2",
+				"Standard_D3",
+				"Standard_DS3",
+				"Standard_D4",
+				"Standard_DS4",
+				"Standard_D11",
+				"Standard_DS11",
+				"Standard_D12",
+				"Standard_DS12",
+				"Standard_D13",
+				"Standard_DS13",
+				"Standard_D14",
+				"Standard_DS14"
+			],
+			"metadata": {
+				"Description": "The size of the SP VMs Created"
+			},
+			"defaultValue": "Standard_DS2"
+		},
+		"adImagePublisher": {
```

Comment:
Take a look at https://github.com/Azure/azure-quickstart-templates/blob/master/101-simple-windows-vm-data-disk/azuredeploy.json
And consider simplifying the parameterization here.